Source/JavaScriptCore/runtime/ArrayPrototype.cpp

```diff
@@ -1511,16 +1512,36 @@ void clearElement(double& element)
     element = PNaN;
 }
 
-template<typename T>
-ALWAYS_INLINE void copyElements(T* buffer, unsigned offset, T* source, unsigned sourceSize, IndexingType sourceType)
+template<typename T, typename U>
+ALWAYS_INLINE void copyElements(T* buffer, unsigned offset, U* source, unsigned sourceSize, IndexingType sourceType)
 {
-    if (sourceType != ArrayWithUndecided) {
-        gcSafeMemcpy(buffer + offset, source, sizeof(JSValue) * sourceSize);
+    if (sourceType == ArrayWithUndecided) {
+        for (unsigned i = sourceSize; i--;)
```

Comment:
Any reason to copy backwards?

Reply:
I think there is no particular reason. We should just make it forward copy, changed.